docs/variants/dell_optiplex/building-manual.md

```diff
@@ -1,53 +1,122 @@
 
 1. Clone the coreboot repository:
 
     ```bash
-    git clone https://review.coreboot.org/coreboot.git
+    git clone https://github.com/dasharo/coreboot.git
     ```
 
-2. Get the submodules:
-
     ```bash
     cd coreboot
-    git submodule update --init --recursive --checkout
     ```
 
-3. Checkout Dasharo branch for OptiPlex 7010/9010 (replace vX.Y.Z with valid
-   version):
+    Replace vX.Y.Z with valid version:
 
     ```bash
-    git remote add dasharo https://github.com/dasharo/coreboot.git
-    git fetch dasharo
     git checkout dell_optiplex_9010_vX.Y.Z
     ```
 
-4. Start docker container:
+    Checkout subdmodules:
+
+    ```bash
+    git submodule update --init --recursive --checkout
+    ```
+
+1. Start docker container:
 
     ```bash
     docker run --rm -it \
        -v $PWD:/home/coreboot/coreboot \
        -w /home/coreboot/coreboot \
-       coreboot/coreboot-sdk:0ad5fbd48d /bin/bash
+       coreboot/coreboot-sdk:VERSION /bin/bash
+    ```
+
+    * `VERSION` should be replaced according to version you building:
+      - `v0.1.0` - `2022-04-04_9a8d0a03db`
+
+1. Inside of the container, configure and start the build process:
+
+    ```bash
+    make distclean
+    ```
+
+    ```bash
+    cp configs/config.dell_optiplex_9010 .config
     ```
 
-5. Inside of the container, configure and start the build process:
+    ```bash
+    make olddefconfig
+    ```
 
     ```bash
-    (docker)make distclean
-    (docker)cp configs/config.dell_optiplex_9010_sff .config
-    (docker)make olddefconfig
-    (docker)make
+    make
     ```
 
     or simply:
 
     ```bash
-    (docker)make distclean && cp configs/config.dell_optiplex_9010_sff .config && make olddefconfig && make
+    make distclean && cp configs/config.dell_optiplex_9010 .config && make olddefconfig && make
+    ```
+
+This will produce a release binary placed in `build/coreboot.rom`. To flash
+coreboot refer to [initial deployment manual](initial-deployment.md).
+
+## Debug build
+
+1. Clone the coreboot repository:
+
+    ```bash
+    git clone https://github.com/dasharo/coreboot.git
+    ```
+
+    ```bash
+    cd coreboot
+    ```
+
+    Replace vX.Y.Z with valid version:
+
+    ```bash
+    git checkout dell_optiplex_9010_vX.Y.Z
+    ```
+
+    Checkout subdmodules:
+
+    ```bash
+    git submodule update --init --recursive --checkout
+    ```
+
+1. Start docker container:
+
+    ```bash
+    docker run --rm -it \
+       -v $PWD:/home/coreboot/coreboot \
+       -w /home/coreboot/coreboot \
+       coreboot/coreboot-sdk:VERSION /bin/bash
+    ```
+
+    * `VERSION` should be replaced according to version you building:
+      - `v0.1.0` - `2022-04-04_9a8d0a03db`
+
+1. Inside of the container, configure and start the build process:
+
+    ```bash
+    make distclean
+    ```
+
+    ```bash
+    cp configs/config.dell_optiplex_9010 .config
+    ```
+
+    ```bash
+    sed -i 's/CONFIG_DEFAULT_CONSOLE_LOGLEVEL_0=y/CONFIG_DEFAULT_CONSOLE_LOGLEVEL_8=y/g' .config
```

Comment:
IIUC, this line is the only change between Debug and Release build. Is it really worth to duplicate the whole section ? We have just proven, that it may easily get out of sync (e.g. we change something in one section, but not in the another).

Reply:
The easiest I can imagine right now is that we have two configs: release and debug. We have a single build procedure and in the configuration step, we instruct to use either debug or release config, depending on the needs.

I can imagine the dbug one will expend - as soon as we enable Tianocore, we will want to enable debug there as well.

Reply:
Makes sense, let me push second config to coreboot and fix documentation here.

Reply:
@macpijan fixed in coreboot https://github.com/Dasharo/coreboot/pull/202/commits/578d5c25b8f6571da37ce766d6bfa5aaa9aed8ca

and in documentation

https://github.com/Dasharo/docs/pull/139/commits/0af80f93765774eed79213c8cf33e84603c449ea

Reply:
@macpijan ping

Reply:
We already maintain targets with multiple configs. Another option is to have a defconfig that does not contain any loglvel/debug settings and append what is needed runtime by choosing release/debug options, however coreboot defaults to high debug level if nothing is specified. At this point I think we may close this discussion.